gpii/node_modules/testing/src/Integration.js

```diff
@@ -369,3 +372,57 @@ fluid.defaults("gpii.test.integration.mockSettingsHandlerRegistry.android", {
         }
     }
 });
+
+fluid.defaults("gpii.test.integration.mockDeviceReportersRegistry", {
+    gradeNames: ["fluid.eventedComponent", "autoInit"],
+    members: {
+        deviceReporters: {}
+    },
+    listeners: {
+        onCreate: "gpii.test.integration.mockDeviceReportersRegistry.populate"
+    }
+});
+
+gpii.test.integration.mockDeviceReporters = {};
+
+gpii.test.integration.mockDeviceReporters.findNameInList = function (expectInstalled, name) {
+    return expectInstalled.some(function (el) { return el === name; });
+};
+
+gpii.test.integration.mockDeviceReportersRegistry.populate = function (that) {
+    fluid.each(that.options.deviceReporters, function (options, key) {
+        var expectInstalled = [];
+        fluid.each(gpii.tests, function (tests) {
```

Comment:
I'm astonished that this works at all! In general we can't assume that everything in gpii.tests.* contains a definition of the expected form, this is a dependency time bomb waiting to go off. You should instead add a specific grade name, for example, gpii.test.deviceReporterAware test into the "gradeNames" entry of your test definitions, and then insert this code into an "onCreate" listener of this grade.